server/src/oauth/db.rs

```diff
@@ -0,0 +1,423 @@
+use super::*;
+use crate::state::AppState;
+use sqlx::PgPool;
+use std::time::{SystemTime, UNIX_EPOCH};
+use uuid::Uuid;
+
+// We need to ensure we're using the actual encryption module
+use self::encryption_helpers::*;
+
+// Create a local module with functions that call the external encryption functions
+mod encryption_helpers {
+    use age::x25519::Identity;
+    use std::sync::Arc;
+
+    pub async fn encrypt(data: &str, key: &Arc<Identity>) -> cja::Result<String> {
+        // Call the encryption function from the main crate
+        crate::encryption::encrypt(data, key).await
+    }
+
+    pub async fn decrypt(encrypted_base64: &str, key: &Arc<Identity>) -> cja::Result<String> {
+        // Call the decryption function from the main crate
+        crate::encryption::decrypt(encrypted_base64, key).await
+    }
+}
+
+/// Stores a new OAuth session in the database
+pub async fn store_session(pool: &PgPool, session: &OAuthSession) -> cja::Result<Uuid> {
+    let session_id = Uuid::new_v4();
+
+    // Store PKCE data and DPoP nonce in the data JSONB field
+    let data = serde_json::json!({
+        "code_verifier": session.code_verifier,
+        "code_challenge": session.code_challenge,
+        "dpop_nonce": session.dpop_nonce
+    });
```

Comment:
Can we encrypt this? If the `data` column is JSONB today we won't be able to easily. It might also not be truely sensitive...

If the column is JSONB lets leave it as is, if not lets encrypt it